File: reflector.py
import string

reflector_codes = {"A": "EJMZALYXVBWFCRQUONTSPIKHGD",
                   "B": "YRUHQSLDPXNGOKMIEBFZCWVJAT",
                   "C": "FVPJIAOYEDRZXWGCTKUQSBNMHL",}

class Reflector:

    def __init__(self, code):
        self.main_dict = dict(zip(string.ascii_lowercase, reflector_codes[code].lower()))

    def reflect(self, letter):
        return self.main_dict[letter]

File: reflector.js
const reflectorCodes = {
  "A": "EJMZALYXVBWFCRQUONTSPIKHGD",
  "B": "YRUHQSLDPXNGOKMIEBFZCWVJAT",
  "C": "FVPJIAOYEDRZXWGCTKUQSBNMHL",
};

class Reflector {
  constructor(code) {
    this.mainDict = {};
    const alphabet = "abcdefghijklmnopqrstuvwxyz";

    for (let i = 0; i < alphabet.length; i++) {
      this.mainDict[alphabet[i]] = reflectorCodes[code][i].toLowerCase();
    }
  }

  reflect(letter) {
    return this.mainDict[letter];
  }
}

module.exports = Reflector;